widget/diagramwidget/connectionpad.go

```diff
@@ -0,0 +1,226 @@
+package diagramwidget
+
+import (
+	"image/color"
+
+	"fyne.io/x/fyne/widget/diagramwidget/geometry/r2"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/canvas"
+	"fyne.io/fyne/v2/driver/desktop"
+	"fyne.io/fyne/v2/widget"
+)
+
+const (
+	pointPadSize float32 = 10
+	padLineWidth float32 = 3
+)
+
+// ConnectionPad is an interface to a connection area on a DiagramElement.
+type ConnectionPad interface {
```

Comment:
Should include an example of using `ConnectionPad` in the demo, unless it's already in there and I just missed it?

Reply:
nevermind, I see it actually is used :)